### PR TITLE
ci: an attempt to run the tests under ASan/UBsan

### PR DESCRIPTION
**Pull request by evverx**
The idea is to run the tests under ASan/UBsan on GHActions when PRs are opened to prevent memory leaks, heap-buffer-overflows and the like from making it into the lxc repository.

### CODE REVIEW

**Comment by evverx** [comment]
Now that it's green I think it should be more or less ready for review. @brauner could you take a look?

**Comment by brauner** [comment]
jenkins: test this please

File: .github/workflows/sanitizers.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+printf "Memory leaks are ignored due to https://github.com/lxc/lxc/issues/3788.\n"
+ASAN_OPTIONS=$ASAN_OPTIONS:detect_leaks=0 UBSAN_OPTIONS=$UBSAN_OPTIONS /usr/bin/test-concurrent.orig
+EOF
+chmod +x /usr/bin/lxc-test-concurrent
+
+if [[ "$CC" == "clang" ]]; then
+    mv /usr/bin/{lxc-test-console-log,test-console-log.orig}
+    cat <<EOF >/usr/bin/lxc-test-console-log
+#!/bin/bash
+printf "Memory leaks are ignored due to https://github.com/lxc/lxc/issues/3796.\n"
```
Comment (brauner) [inline .github/workflows/sanitizers.sh:55]: This should hopefully be fixed with:
https://github.com/lxc/lxc/pull/3790

**Comment by brauner** [comment]
Looks good to me. I think we want to move all the ASAN options (including compiler dependent ones) into the makefile itself in the future.

File: .github/workflows/sanitizers.sh (hunk 1 above)
Comment (evverx) [reply .github/workflows/sanitizers.sh:55]: I tried rebasing this PR on top of the master branch (with those commits included) but the test built with clang somehow triggered the same memory leak again. Weirdly, gcc seems to be fine and I can't seem to reproduce it locally with the same version of clang.

**Pull request merged**